controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go

```diff
@@ -495,7 +508,7 @@ func (in *KubeadmControlPlane) validateCoreDNSVersion(prev *KubeadmControlPlane)
 			field.Invalid(
 				field.NewPath("spec", "kubeadmConfigSpec", "clusterConfiguration", "dns", "imageTag"),
```

Comment:
Could we add the path prefix as a parameter in this function too?

Reply:
+1

Reply:
Reconsidering. This function is called on the whole KubeadmControlPlane (not on a sub-struct). It is also only used for KubeadmControlPlane (KubeadmControlPlaneTemplates are immutable).

I would keep it as is.